app/services/payment_requests/payments/flutterwave_service.rb

```diff
@@ -0,0 +1,208 @@
+# frozen_string_literal: true
+
+module PaymentRequests
+  module Payments
+    class FlutterwaveService < BaseService
+      include Customers::PaymentProviderFinder
+      include Updatable
+
+      def initialize(payable:)
+        @payable = payable
+
+        super
+      end
+
+      def call
+        result.payment_url = payment_url
+        result
+      rescue LagoHttpClient::HttpError => e
+        deliver_error_webhook(e)
+
+        result.service_failure!(code: "action_script_runtime_error", message: e.message)
+      end
+
+      def update_payment_status(organization_id:, status:, flutterwave_payment:)
+        payment = if flutterwave_payment.metadata[:payment_type] == "one-time"
+          create_payment(flutterwave_payment)
+        else
+          Payment.find_by(provider_payment_id: flutterwave_payment.id)
+        end
+        return result.not_found_failure!(resource: "flutterwave_payment") unless payment
+
+        result.payment = payment
+        result.payable = payment.payable
+        return result if payment.payable.payment_succeeded?
+
+        payment.status = status
+
+        payable_payment_status = payment.payment_provider&.determine_payment_status(payment.status)
+        payment.payable_payment_status = payable_payment_status
+        payment.save!
+
+        update_payable_payment_status(payment_status: payable_payment_status)
+        update_invoices_payment_status(payment_status: payable_payment_status)
+        update_invoices_paid_amount_cents(payment_status: payable_payment_status)
+        reset_customer_dunning_campaign_status(payable_payment_status)
+
+        PaymentRequestMailer.with(payment_request: payment.payable).requested.deliver_later if result.payable.payment_failed?
+
+        result
+      rescue ActiveRecord::RecordInvalid => e
+        result.record_validation_failure!(record: e.record)
+      rescue BaseService::FailedResult => e
+        result.fail_with_error!(e)
+      end
+
+      private
+
+      attr_reader :payable
+
+      delegate :organization, :customer, to: :payable
+
+      def payment_url
+        response = create_checkout_session
+
+        response["data"]["link"]
+      end
+
+      def create_checkout_session
+        body = {
+          amount: payable.total_amount_cents / 100.0,
```

Comment:
Same comment than with the `invoice` implementation. Happy to help on this if you need